tests/unit/app/models/test_config.py

```diff
@@ -1086,3 +1086,157 @@ def test_config_improper_model():
                 },
             }
         ).validate_yaml()
+
+
+def test_config_improper_question_filter():
+    """Test the Config model with improper question filter (no name) is set."""
+    with pytest.raises(
+        InvalidConfigurationError,
+        match="name, pattern and replace_with need to be specified",
+    ):
+        Config(
+            {
+                "llm_providers": [
+                    {
+                        "name": "openai",
+                        "url": "http://test_provider_url",
+                        "credentials_path": "tests/config/secret.txt",
+                        "models": [
+                            {
+                                "name": "test_model_name",
+                                "url": "http://test_model_url",
+                                "credentials_path": "tests/config/secret.txt",
+                            }
+                        ],
+                    }
+                ],
+                "ols_config": {
+                    "default_provider": "openai",
+                    "default_model": "test_default_model",
+                    "classifier_provider": "test_classifer_provider",
+                    "classifier_model": "test_classifier_model",
+                    "summarizer_provider": "test_summarizer_provider",
+                    "summarizer_model": "test_summarizer_model",
+                    "validator_provider": "test_validator_provider",
+                    "validator_model": "test_validator_model",
+                    "question_filters": [
+                        {
+                            "pattern": "test_regular_expression",
+                            "replace_with": "test_replace_with",
+                        }
+                    ],
+                    "conversation_cache": {
+                        "type": "memory",
+                        "memory": {
+                            "max_entries": 100,
+                        },
+                    },
+                    "logging_config": {
+                        "app_log_level": "error",
+                    },
+                },
+            }
+        ).validate_yaml()
+
+
+def test_config_with_multiple_question_filter():
+    """Test the Config model with multiple question filter is set."""
+    with pytest.raises(
+        InvalidConfigurationError,
+        match="default_model specifies an unknown model test_default_model",
```

Comment:
this doesn't seem like the right expected error.  Is it supposed to be an error to have 2 filters with the same name?  if so this should expect that error.

If the filters defined here are supposed to be considered valid, then fix the test so that the config is overall valid and no exception is expected.